PostHog Array 1.38.0 (#3826)
PostHog/posthog.com: contents/blog/the-posthog-array-1-38-0.mdx

### Improved: <Highlight 4>

Please add a screenshot here.
Please add a sentence of copy about this highlight.

### Improved: <Highlight 5>

Please add a screenshot here.
Please add a sentence of copy about this highlight.

### Other improvements & fixes

Version 1.37 also adds hundreds of other improvements and fixes, including...

-   **Fixed**: 
-   **Fixed**: 
-   **Fixed**: 
-   **Fixed**: 
-   **Fixed**: 

View the commit log in GitHub for a full history of changes: [`release-1.37.0...release-1.38.0`](https://github.com/PostHog/posthog/compare/release-1.37.0...release-1.38.0).

### Deprecation and removal notices

-  Please add any deprecation notices. 

## Give us your feedback

We’re always working on improving PostHog and would love to talk to you! Please [schedule a 30 minute call](https://calendly.com/posthog-feedback) with one of our Product, Engineering, or Marketing team members to help us understand how to improve. As a thank you for your time, we'll be giving away awesome [PostHog merch](https://merch.posthog.com)!

## Contributions from the community

We always welcome contributions from our community and this time we want to thank the following people...

> <Joe to add contributors here>

Reviewer: https://github.com/PostHog/posthog.com/pull/3776 is worth adding!

### Improved: Funnel Breakdown Attribution

You can now choose which specific funnel step a breakdown property should come from, and if the same value should be copied over to other funnel steps or not for analysis. [Read more here](https://posthog.com/docs/user-guides/funnels#choosing-breakdown-property-behaviour)

### New: Four new apps released

The community has been busy building many new apps for the PostHog app store and we're excited to announce the following apps have been released for users on PostHog Cloud...

- [Variance Connector](/apps/variance-connector), for sending event data to [Variance](https://www.variance.com/) so you can define accurate PQLs
- [Unduplicator](/apps/unduplicator), for cleaning duplicate events during data imports
- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending user data to [Engage](https://engage.so/) for segmentation and marketing automation

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Other improvements & fixes

Version 1.37 also adds hundreds of other improvements and fixes, including...

> -   **Fixed**: 

Reviewer 2: ```suggestion
-   **Actions have been renamed to Calculated Events**: Actions are now called calculated events and can be found and created from the Data Management Events tab.
```

For posthog-js only for now, but you can ensure that users get a consistent experience with feature flags across login steps. This is great for experiments where you'd want the same user to be shown the same variant throughout, no matter how their userID changes. [Read more](https://posthog.com/docs/user-guides/feature-flags#persisting-feature-flags-across-authentication-steps)

### Improved: Funnel Breakdown Attribution

You can now choose which specific funnel step a breakdown property should come from, and if the same value should be copied over to other funnel steps or not for analysis. [Read more here](https://posthog.com/docs/user-guides/funnels#choosing-breakdown-property-behaviour)

### Improved: Libraries support for multivariate feature flags and experiments

Our Node, Ruby, Go, and PHP libraries now have support for experiments! In beta right now is also feature flags, groups, and session analytics support for our mobile libraries.

### New: Four new apps released

The community has been busy building many new apps for the PostHog app store and we're excited to announce the following apps have been released for users on PostHog Cloud...

> - [Variance Connector](/apps/variance-connector), for sending event data to [Variance](https://www.variance.com/) so you can define accurate PQLs

Reviewer: ```suggestion
- [Variance Connector](/apps/variance-connector), for sending event data to [Variance](https://www.variance.com/) so you can define accurate Product Qualified Leads
```

- [Unduplicator](/apps/unduplicator), for cleaning duplicate events during data imports
- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending user data to [Engage](https://engage.so/) for segmentation and marketing automation

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Other improvements & fixes

Version 1.37 also adds hundreds of other improvements and fixes, including...

-   **Fixed**: 

> -   **Fixed**: 

Reviewer 3: ```suggestion
-   **Fixed**: Update to posthog-js and Hubspot app making sure we don't override `initial_referrer` and `initial_referring_domain` . If you ever saw these change and not reflect the true initial values update your posthog-js version (& hubspot app on self-hosted) to make sure we don't override them in the future.
```

> -   **Fixed**: 

Reviewer 3: ```suggestion
-   **Improvement**:  Toolbar won't show up automatically anymore, but you can still launch it from the app. This made calls to feature flags faster too :)
```

### Improved: Library support for multivariate feature flags and experiments
Our Node, Ruby, Go, and PHP libraries have been updated to support experiments! We are also trialling support for feature flags, groups, and session analytics in our mobile libraries. Interested in giving it a go? Let us know in [the Slack community](/slack)!

### New: Four new apps released
The community has been busy building many new apps for the PostHog app store and we're excited to announce the following apps have been released for users on PostHog Cloud...

- [Unduplicator](/apps/unduplicator), for cleaning duplicate events during data imports
- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending user data to [Engage](https://engage.so/) for segmentation and marketing automation
- [Variance Connector](/apps/variance-connector), for sending event data to [Variance](https://www.variance.com/) so you can define accurate Product Qualified Leads in your CRM

Interested in building your own app? [Here's how to get started](/docs/apps)!

> 

Reviewer 3: ```suggestion
### Improved: Optimized ClickHouse space and speeding up queries

`properties` column takes up most of the space in any PostHog ClickHouse setup. This release changes out the default compression scheme for ZSTD(3). This reduces space usage by around 2.4x and can speed up uncached queries up to 2.4x.

See https://github.com/PostHog/posthog/issues/10616 for full measurements and other details.

After running this migration only new data will use the updated compression scheme. To get the full benefit of compression on existing data, run `OPTIMIZE TABLE sharded_events FINAL` on your clickhouse cluster which will rewrite existing data.
```

---
date: 2022-06-28
title: 'Array 1.38.0: Exports, subscriptions and session analysis!'
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.38.0 introduces exports and subscriptions for dashboards and insights, plus improved feature flag persistence and much more!
---

import SessionAnalysis from '../images/blog/array/1-38-0-session-analysis.png'
import BreakdownBins from '../images/blog/array/1-38-0-breakdown.png'
import SearchImprove from '../images/blog/array/1-38-0-search.png'

> Want to know more about what we're up to? [Subscribe to HogMail, our new newsletter](/newsletter), which we send out every two weeks!

Reviewer 4: ```suggestion
Want to know more about what we're up to? [Subscribe to HogMail, our newsletter](/newsletter), which we send out every two weeks!
```

Universal search wasn't the only search-based update this time, either. We also made searching on the Persons page _much_ faster and more reliable than it was before. 

### Improved: Funnel breakdown attribution
You can now choose which specific funnel step a breakdown property should come from, and whether the same value should be copied over to other funnel steps for analysis. This is very handy for getting even more value from funnels, and further information is included in the [funnel documentation](https://posthog.com/docs/user-guides/funnels#choosing-breakdown-property-behaviour).

### Improved: Library support for multivariate feature flags and experiments
Our Node, Ruby, Go, and PHP libraries have been updated to support experiments! We are also trialling support for feature flags, groups, and session analytics in our mobile libraries. Interested in giving it a go? Let us know in [the Slack community](/slack)!

### New: Four new apps released

> The community has been busy building many new apps for the PostHog app store and we're excited to announce the following apps have been released for users on PostHog Cloud...

Reviewer 4: ```suggestion
The community has been busy building many new apps for the [PostHog App Store](/apps) and we're excited to announce the following apps have been released for users on PostHog Cloud...
```

- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending data to [Engage](https://engage.so/) for marketing automation
- [Variance Connector](/apps/variance-connector), for sending data to [Variance](https://www.variance.com/) so you can define PQLs

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Improved: Optimized ClickHouse space and speeding up queries
The `properties` column takes up most of the space in any PostHog ClickHouse setup. [This release](https://github.com/PostHog/posthog/issues/10616) changes out the default compression scheme for ZSTD(3), reducing space by around 2.4x and speeding up uncached queries by up to 2.4x.

After running this migration only new data will use the updated compression scheme. To get the full benefit of compression on existing data, run `OPTIMIZE TABLE sharded_events FINAL` on your clickhouse cluster which will rewrite existing data.

### Other improvements & fixes
Version 1.38 also adds hundreds of other improvements and fixes, including...

> -   **Actions have been renamed to Calculated Events**: Actions are now called calculated events and can be found and created from the Data Management Events tab.

Reviewer 4: Just leaving a note here in case we change our minds about this?

> -   **Fixed**: Update to posthog-js and Hubspot app making sure we don't override `initial_referrer` and `initial_referring_domain` . If you ever saw these change and not reflect the true initial values update your posthog-js version (& hubspot app on self-hosted) to make sure we don't override them in the future.

Reviewer 4: ```suggestion
-   **Fixed**: Update to posthog-js and Hubspot app making sure we don't override `initial_referrer` and `initial_referring_domain`. If you ever saw these change and not reflect the true initial values update your posthog-js version (& Hubspot app on self-hosted) to make sure we don't override them in the future.
```

-   **Improvement**:  Toolbar won't show up automatically anymore, but you can still launch it from the app. This made calls to feature flags faster too :)
-   **Improvement**: We now automatically comment all SQL queries generated by PostHog with tags. This helps track down performance issues on self-hosted instances. 

> -   **Breaking change**: `onAction` plugin server function is removed. Improving the efficiency of the plugin-server.

Reviewer 5: ```suggestion
-   **Breaking change**: `onAction` plugin server function is removed. Improving the efficiency of the plugin-server.
-   **Improvement**: CSV and Image exports now use [Object Storage when it is enabled](https://posthog.com/docs/self-host/deploy/configuration#minio)
```

View the commit log in GitHub for a full history of changes: [`release-1.37.0...release-1.38.0`](https://github.com/PostHog/posthog/compare/release-1.37.0...release-1.38.0).

## Give us your feedback
We’re always working on improving PostHog and would love to talk to you! Please [schedule a 30 minute call](https://calendly.com/posthog-feedback) with one of our Product, Engineering, or Marketing team members to help us understand how to improve. As a thank you for your time, we'll be giving away awesome [PostHog merch](https://merch.posthog.com)!

## Contributions from the community
We always welcome contributions from our community and this time we want to thank the following people...

- [@Sepal](https://github.com/sepal), for adding [UTM segmentation docs](/docs/user-guides/utm-segmentation)

> - [@Zeviset](https://github.com/zevisert), for correction some GitLab Auth secrets

Reviewer 4: ```suggestion
- [@Zeviset](https://github.com/zevisert), for correcting some GitLab Auth secrets
```

---
date: 2022-06-28

> title: 'Array 1.38.0: Exports, subscriptions and session analysis!'

Author: ```suggestion
title: 'Array 1.38.0: Exports, subscriptions and session analysis'
```

- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending data to [Engage](https://engage.so/) for marketing automation
- [Variance Connector](/apps/variance-connector), for sending data to [Variance](https://www.variance.com/) so you can define PQLs

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Improved: Optimized ClickHouse space and speeding up queries
The `properties` column takes up most of the space in any PostHog ClickHouse setup. [This release](https://github.com/PostHog/posthog/issues/10616) changes out the default compression scheme for ZSTD(3), reducing space by around 2.4x and speeding up uncached queries by up to 2.4x.

After running this migration only new data will use the updated compression scheme. To get the full benefit of compression on existing data, run `OPTIMIZE TABLE sharded_events FINAL` on your clickhouse cluster which will rewrite existing data.

### Other improvements & fixes
Version 1.38 also adds hundreds of other improvements and fixes, including...

> -   **Actions have been renamed to Calculated Events**: Actions are now called calculated events and can be found and created from the Data Management Events tab.

Author: ```suggestion
```

---
date: 2022-07-01
title: 'Array 1.38.0: <Joe to add headline here>'
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
categories: ['Release notes', 'Product updates']
featuredImage: ../images/blog/posthog-array-blog.png
featuredImageType: full
excerpt: PostHog 1.38.0 introduces <Joe to add detail here>
---

import SessionAnalysis from '../images/blog/array/1-38-0-session-analysis.png'
import BreakdownBins from '../images/blog/array/1-38-0-breakdown.png'

Want to know more about what we're up to? [Subscribe to HogMail, our new newsletter](/newsletter), which we send out every two weeks!

> Running a self-hosted instance? Check out our [Upgrading PostHog guide](/docs/self-host/configure/upgrading-posthog).

## PostHog 1.38.0 release notes

[ANY IMPORTANT DEPRECATION NOTICES CAN GO HERE]

**Release highlights:**

- New: [Highlight 1]
- New: [Highlight 2]
- New: [Highlight 3]
- Improved: [Highlight 4]
- Improved: [Highlight 5]

> ### New: [Highlight 1]

Reviewer 3: ```suggestion
### New: [Async migration 0005](https://github.com/PostHog/posthog/blob/master/posthog/async_migrations/migrations/0005_person_replacing_by_version.py)

There is a new async migration available, that changes the persons table in ClickHouse to use `version` column instead of `_timestamp` for collapsing entires. This helps avoid various data integrity issues due to race conditions and batching of Kafka messages within plugin-server.

More info about async migrations: https://posthog.com/docs/self-host/configure/async-migrations/overview
More info about async migration 0005: https://github.com/PostHog/posthog/blob/master/posthog/async_migrations/migrations/0005_person_replacing_by_version.py
```

Our Node, Ruby, Go, and PHP libraries now have support for experiments! In beta right now is also feature flags, groups, and session analytics support for our mobile libraries.

### New: Four new apps released

The community has been busy building many new apps for the PostHog app store and we're excited to announce the following apps have been released for users on PostHog Cloud...

- [Variance Connector](/apps/variance-connector), for sending event data to [Variance](https://www.variance.com/) so you can define accurate Product Qualified Leads in your CRM
- [Unduplicator](/apps/unduplicator), for cleaning duplicate events during data imports
- [URL Normalizer](/apps/url-normalizer), for introducing consistent casing across PostHog 
- [Engage Connector](/apps/engage-connector), for sending user data to [Engage](https://engage.so/) for segmentation and marketing automation

Interested in building your own app? [Here's how to get started](/docs/apps)!

### Other improvements & fixes

Version 1.37 also adds hundreds of other improvements and fixes, including...

-   **Fixed**: 
-   **Fixed**: 
-   **Fixed**: 

> -   **Fixed**: 

Reviewer 3: ```suggestion
-   **Improvement**: We now automatically comment all SQL queries generated by PostHog with tags. This helps track down performance issues on self-hosted instances. 
```

> -   **Fixed**: 

Reviewer 3: ```suggestion
-   **Breaking change**: `onAction` plugin server function is removed. Improving the efficiency of the plugin-server.
```
cc @macobo
